File: src/main/java/dashflight/sparkbootstrap/Location.java
package dashflight.sparkbootstrap;

public class Location {

    private int id;
    private String name;

    public Location(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public static Location withFields(Integer id, String name) {
        if (id == null || name == null) {
            return null;
        }

        return new Location(id, name);
    }
}

File: src/main/java/dashflight/sparkbootstrap/Location.kt
package dashflight.sparkbootstrap

class Location(val id: Int, val name: String) {

    companion object {
        fun withFields(id: Int?, name: String?): Location? {
            return if (id == null || name == null) {
                null
            } else Location(id, name)
        }
    }

}